perceval/backends/github.py

```diff
@@ -111,13 +137,41 @@ def fetch_from_cache(self):
 
         cache_items = self.cache.retrieve()
 
-        for raw_issues in cache_items:
+        # In the cache first is the item with issues and next users data
```

Comment:
I have another version here. It checks some possible failures that you haven't taken into account. I think it's more clear what it does. I didn't check it but I think it should work as is:

```python
issues = None

while True:
    try:
        raw_item = next(cache_items)
    except StopIteration:
        if issues:
            for issue in self._build_issues(issues):
                yield issue
        break

    item = json.loads(raw_item)

    if 'login' in item:
        try:
            raw_orgs = next(cache_items)
        except StopIteration:
            # Fatal error. Cache should had stored an organizations item
            # per parsed user.
            cause = "cache is exhausted but more items were expected"
            raise CacheError(cause=cause)

        item['organizations'] = json.loads(raw_orgs)
        self._users[item['login']] = item
        continue

    # A new set of issues has been read. It means we already
    # have the enough information to build and return the
    # previous set
    if issues:
        for issue in self._build_issues(issues):
            yield issue

    # Next issues to parse
    issues = item

def _build_issues(self, issues):
    for issue in issues:
        for field in ['user', 'assignee']:
            issue[field + '_data'] = {}

            if issue[field]:
                issue[field + '_data'] = \
                    self._users[issue[field]['login']]
        yield issue
```